Gate 2

pkurras opened the pull request:
This is the final version for Gate 2

pkurras committed:
Mirror changes

[*] mirror reflection is now independent from a pivot object (it uses the forward vector as the reflection normal)

darktoastbrot committed:
Boss Phase Implementet

[+] Phase System
[+] Phase 1 (mirror)

darktoastbrot committed:
Boss Phase Mirror

[+]Mirror Phase fixed

darktoastbrot committed:
Mirror Main Fixes

[+] Reflect on other Mirrors
[+] Laser Attack Enemy and Player

pkurras committed:
Drop Component

[+] added drop component
[+] added spawn behaviour

darktoastbrot committed:
Boss Phase Controller

[+] Boss Phase Controller

pkurras committed:
Spear spawner fix

[+] added spear prefab (using heart rn)
[*] spear spawner now uses the width of the box collider 2d (image width didn't work)

pkurras committed:
Spear functionality

[+] added spear component

[+] added countdown utility class

[*] Spear can drop now

pkurras committed:
Spear functionality #2

[!] forgot to add actual spear class and countdown ... did that now. R.I.P.

pkurras committed:
Jump variable fix & Drop addition

[+] added damage to spears and also respawn
[*] jump variables should be public now (handled by velocity)
[-] removed the logging from countdown

pkurras committed:
Music Zone

[+] added functionality for music zones
[+] added boss room music zone
[-] removed the audio source from the tilemap and added it to the player

pkurras committed:
Knockback Component

[+] added knockback component
[!] PLAYER NEEDS HEAVY CLEANUP NOW

Helero committed:
Gate 2 lv

[+]new scene for gate 2
[*] changed character sprite
[*] changed boss sprite
[*]testet mirror system
[*] fixed mirror phase turn of boss

darktoastbrot committed:
Enemy Damage Fix

[*] fixed player making damage to enemy

darktoastbrot committed:
Mirror Damage Fix 

[*]mirror can make damage again

pkurras committed:
Various changes (collected)

[+] test scene
[*] boss phase working again. pls revert boss prefab in Gate2 scene
[*] mirror working now fully in Gate2 scene
[*] in Gate2 scene changed position of the mirrors
[*] level design changes 
[*] QA phase test

darktoastbrot committed:
Laser fixes

[-]Double Player Shooting Component removed
[*] Player rotation fix

Helero committed:
LevelDesign

[*]changed position of mirrors
[*] changed position of platforms
[+]added corridor

darktoastbrot committed:
Boss Phase Update

[+] phases can be combined
[+] Phase Manager to set phase system

darktoastbrot committed:
Added Phases in Gate2Scene

[+] dragged some objects in the scene

Helero committed:
Level design changes

[*] changed levellayout
[*]changed boss phase variablen
[*] changed player variablen

darktoastbrot committed:
Boss Phases added

[+] 2 new Phases (Fireball, Charge)
[+] Camera follow script
[+] Spear Phase
[*] Knockback fix

pkurras committed:
Camera Clipping

[+] camera now clips to level bounds

darktoastbrot committed:
Phase Controller and laser fix

[*] Phase controller fix
[*] laser fix

darktoastbrot committed:
Camera Follow

[*] Camera follow better now

darktoastbrot committed:
Many Fixes

[+] player Animations
[+] many many fixes

darktoastbrot committed:
Scene background

[+] Scene background

Helero committed:
Phasen Implementation

[*] chanaged phases
[*]changed mirror position
[+]Animatet Boss phase mirror

darktoastbrot committed:
Bug fixes

[*] fixed ammo is empty visual feetback bug
[*] fixed Shoot goes invisible after it gets reflected bug
[*] fixed Boss Phase Fireball is missing a fireball bug

darktoastbrot committed:
Scene update

[*] Bug fixes in scene

pkurras committed:
Knockback fix

[*] presumably fixed knockback

darktoastbrot committed:
Boss Phase Animation

[+] can add animations in phase controllers by string

pkurras committed:
Final knockback fix

[*] fixed knockback component (works now as intended)

Helero committed:
GameDesign

[*]level design changes
[*]mirrorplacementchanges
[*]small balancing changes

Helero committed:
Boss phase

[+]added the boss phase animations
[*]changed boss balance
[*] changed enviourment

Helero committed:
Boss balancing changes

[*] changed boss balancing

darktoastbrot committed:
Duck fix and camera fix

[*] fixed duck
[*] camera FIX

Helero committed:
Boss GameDesign

[+] added other spride for boss
[*]changed boss prefab hitbox
[*]changed phase manager balancing

darktoastbrot committed:
Camera fix

[*] re-wrote the values in CameraFollow.cs to fix it

darktoastbrot committed:
Spear Phase hitboxes fix

[*] In spear phase, boss can only be hit on his head

darktoastbrot committed:
Boss Contact Damage and Knockback

[+] Implementet boss contact damage. Please Revert Enemy-Prefab and Player-Prefab in the scene, before doing changes. (button "Revert" is on top of the Inspector)

darktoastbrot committed:
Some changes

[*] player can shoot every direction again
[*] some stuff hardcoded, so it will avoid errors when reverting the prefab.

Helero committed:
Boss Balancing

[*] boss phase changes
[*] changed speerphase
[*]changed boss variables

darktoastbrot committed:
Scene Updated

[+] dragged some stuff in

Helero committed:
Animation 

[*]changed animation to batter fitt the phases

darktoastbrot committed:
Spike bug fix

[+] bug fixed

darktoastbrot committed:
fixes

[*] max health now affects the hearts shown in the UI correctly
[*] spear phase is ending again.

pkurras committed:
Fixed protection level

[*] protection level of health fixed

Helero committed:
Boss balancing

[*] boss balancing changes

darktoastbrot committed:
Healthbar fix

[*] healthbar setting the health variable to maxhealth on spawn

Helero committed:
Boss balancing changes

[*]changed variable from boss
[*]changed player health

darktoastbrot committed:
Boss Rotation, Spike position fix

[*] Boss rotating in every phase 
[*] Spike drop boundings updated

Helero committed:
Boss balancing

[*] changed spike position
[*] changed spike values for balancing purpoise

darktoastbrot committed:
Phases fix

[*] fixed bug, that phase stays

Helero committed:
Boss balancing

[+] added enviourment
[+] added boss sprites
[*]changed boss variables

Helero committed:
Enviourment changes

[+]added enviourment assets
[*] changed eviourment assets
[*]changed color of trine

darktoastbrot committed:
Some bug fixes and changes

[+] boss phase start on trigger
[*] mirrorphase+spearphase combination fix
[*]fireballs can shoot through platforms again

darktoastbrot committed:
Boss Room trigger put in scene

[+] boss room trigger

darktoastbrot committed:
Spike ceiling fix

[*] spike not colliding with the ceiling anymore

Helero committed:
boss balancing changes

[*]chaged some boss variables
[+] added ceiling tag

darktoastbrot committed:
trigger deactivating; invincibility

[*] when invincible, knockback still gets applied
[*] Boss Trigger deactivate on enter

pkurras committed:
Status Menu

[+] added status menu

pkurras committed:
Win text

[+] added win text

Helero committed:
spike phase changes

[*] changed sike variables
[*] chnaged game manager script from gate 1 to gate 2

pkurras committed:
Bullet charge

[*] bullet charges now before starting (it has a charge time)

Helero committed:
Spikes

[*]changed spike variables
[*]changed pushback

darktoastbrot committed:
Player damage fix

[*] fixed player damage

pkurras committed:
bullet transform to enemy

[*] enemy reference can be set in bullet class now

darktoastbrot committed:
spear and bullet charge fix

[*] spear phase and bullet charge fixed

darktoastbrot committed:
Fireball sprite changed

[+] fireball sprite animation

Helero committed:
Changed assets

[+]added gate2 final assets
[*]made some variable balancing changes

darktoastbrot committed:
Bullet fix

[*] merge conflict solved perfectly

darktoastbrot committed:
gameOverScreen buttons function added

[*]fdsfg

Helero committed:
fixes

Helero committed:
collider

[+]added box colider for glass cabinets

darktoastbrot committed:
button fix

[*] restart btn working
[*] exit btn working

darktoastbrot committed:
spike fix

[*] spike not moving after dead

Helero committed:
liddlefixes

darktoastbrot committed:
Phases fix

[*] Game can start from scene
[*] phase controller performance fix

ProducerPhil reviewed (approved):
Worked on Gate 2, should work now.

darktoastbrot committed:
Camera Update, Player Shooting fix

[+] Camera gets new left X bound, when Player enters boss room
[*] Player standard shooting direction is looking direction

Helero committed:
Tutorial

[+] added tutorial lv
[*] changed hitbox of boss and fireball
[*]small balancing changes to boss variables

darktoastbrot committed:
Pause Menu + Tutorial Object to destroy

[+] pause menu code
[*] object to destroy in tutorial added

ProducerPhil committed:
Menu Assets

[+] Menu_Credits_DieNerdbeeren
[+] Menu_Options_Slider
[+] Menu_Options_SliderBar
[+] Menu_Options_SliderFrame
[+] Menu_Options_ToggleOff
[+] Menu_Options_ToggleOn
[+] Menu_Title_spr1
[+] Menu_Title_spr2
[+] Menu_Title_spr3
[+] Startscreen_Title_spr1
[+] Startscreen_Title_spr2
[+] Startscreen_Title_spr3

Helero committed:
Tuorial

[+]added enviourment to tutorial
[*] changed layout of tutorial

Helero committed:
tutorial

[*]changed tutorial balancing
[*]fixed mirror bug in tutorial

darktoastbrot committed:
Tutorial Scene fix

[*] scene working now great again
